packages/expo-video/src/VideoView.tsx

```diff
@@ -84,3 +90,43 @@ function getPlayerId(player: number | VideoPlayer): number | null {
   }
   return null;
 }
+
+/**
+ * Returns a shared object, which is automatically cleaned up when the component is unmounted.
+ */
+// TODO: when SharedObject type is added make T extend it
+export function useReleasingSharedObject<T>(factory: () => T, dependencies: DependencyList): T {
+  const objectRef = useRef<any | null>(null);
+  const isFastRefresh = useRef(false);
+  const previousDependencies = useRef<DependencyList>(dependencies);
+
+  if (objectRef.current == null) {
+    objectRef.current = factory();
+  }
+
+  const object = useMemo(() => {
+    // If the dependencies have changed, release the previous object and create a new one, otherwise this has been called
+    // because of a fast-refresh and we don't want to release the object.
+    if (!previousDependencies.current.every((value, index) => value === dependencies[index])) {
+      objectRef.current.release();
```

Comment:
Doesn't it need a `?` after `current`? It can be `null`

Reply:
Yes, typescript didn't show an error because there was an incorrect type in objectRef